src/main/java/duke/maincomponents/Parser.java

```diff
@@ -0,0 +1,203 @@
+                String[] arr = userInput.substring(length + 1).split(" /by ");
+                String deadlineDescription = arr[0];
+                String descriptionBy = arr[1];
+
+                ArrayList<String> tempArr = new ArrayList<>();
+                tempArr.add(deadlineDescription);
+                tempArr.add(descriptionBy);
+
+                return tempArr;
+            } else {
+                throw new DukeException("Error! Please provide when the deadline will be due after /by. "
+                        + "[eg. deadline submit work /by 2020-03-20]");
+            }
+        }
+    }
+
+    public boolean equalsToFind(String userInput) {
+        return checkStringStartingEquals(userInput, "find");
+    }
+
+    public String parseFindCommand(String userInput) throws DukeException {
+        int length = "find".length();
+
+        if (checkStringEquals(userInput, "find")) {
+            throw new DukeException("Error! The description of what you wish to find cannot be empty!");
+        } else if (!checkStringEquals(userInput.substring(length, length + 1), " ")) {
+            throw new DukeException("Error! Please include a space after the string you wish to find.");
+        } else {
+            return userInput.substring(length + 1);
+        }
+    }
+
+    public boolean checkIfExit(String userInput) {
+        return userInput.equals("bye");
+    }
+
+    public Command parse(String userInput) throws DukeException {
+        try {
+            if (this.equalsToList(userInput)) {
+                return new ShowListCommand();
+
+            } else if (this.equalsToDone(userInput)) {
+                int taskDoneInt = this.parseDoneCommand(userInput);
+                return new DoneCommand(taskDoneInt);
+
+            } else if (this.equalsToDelete(userInput)) {
+                int taskDeleteInt = this.parseDeleteCommand(userInput);
+                return new DeleteCommand(taskDeleteInt);
+
+            } else if (this.equalsToToDo(userInput)) {
+                String toDoDescription = this.parseToDoCommand(userInput);
+                return new ToDoCommand(toDoDescription);
+            } else if (this.equalsToEvent(userInput)) {
+                ArrayList<String> eventDescription = this.parseEventCommand(userInput);
+                return new EventCommand(eventDescription);
+            } else if (this.equalsToDeadline(userInput)) {
+                ArrayList<String> eventDescription = this.parseDeadlineCommand(userInput);
+                return new DeadlineCommand(eventDescription);
+            } else if (this.equalsToFind(userInput)) {
+                String stringToFind = this.parseFindCommand(userInput);
+                return new FindCommand(stringToFind);
+            } else {
+                throw new DukeException("I'm sorry, but I don't know what that means");
+            }
+
+        } catch (DukeException e) {
+            throw e;
+        }
+    }
```

Comment:
I think you can remove the lines before each else if statements.

Example:

if() {
    return;
} else {
    return;
}

Reply:
yes your recommendation would definitely make my code cleaner